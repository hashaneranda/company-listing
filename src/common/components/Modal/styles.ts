import styled from 'styled-components';

import { Modal, Paper } from '@material-ui/core';

export const ModalComp = styled(Modal)`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const PaperWrapper = styled(Paper)`
  position: relative;
  padding-top: 3em;
  background: ${props => props.theme.palette.background.secondary};
  color: ${props => props.theme.palette.typography.main};

  &:focus {
    outline: none;
  }
`;

export const Image = styled.img.attrs(props => ({
  src: props.src,
}))`
  position: absolute;
  right: 32px;
  top: 28px;

  width: 15px;
  height: 15px;

  &:hover {
    cursor: pointer;
    transform: scale(1.05);
  }
`;
