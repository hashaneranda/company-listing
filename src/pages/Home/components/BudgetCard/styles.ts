import styled from 'styled-components';

export const Container = styled.div`
  padding: 1em 2em;
  display: grid;
  grid-template-columns: minmax(250px, 25%) 1fr;
`;

export const LeftContainer = styled.div`
  padding: 1em 2em;
  display: grid;
  place-items: center;
`;

export const RightContainer = styled.div`
  padding: 1em 2em;
  display: flex;
  flex-direction: column;
`;

export const BudgetRow = styled.div`
  display: flex;
  flex-direction: row;
`;
