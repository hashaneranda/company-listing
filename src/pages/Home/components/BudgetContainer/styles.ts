import styled from 'styled-components';

export const Card = styled.div`
  padding: 1em 2em;
  display: grid;
  grid-template-columns: minmax(50px, 25%) 1fr;

  &.primary {
    display: flex;
    flex-direction: row;
    justify-content: center;
    gap: 3em;

    h4 {
      font-size: 28px;
      color: ${props => props.theme.palette.color.primary};
      margin-top: 0.2em;
    }

    img {
      width: 55px;
    }
  }
`;

export const CardWrapper = styled.div`
  display: grid;
  place-items: center;

  img {
    width: 35px;
  }
`;

export const CardBody = styled.div`
  display: flex;
  flex-direction: column;

  color: ${props => props.theme.palette.typography.secondary};

  p {
    color: ${props => props.theme.palette.typography.secondary};
  }

  h4 {
    color: ${props => props.theme.palette.typography.main};
  }
`;
