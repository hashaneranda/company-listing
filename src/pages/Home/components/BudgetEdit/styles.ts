import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 40em;
  padding: 1em 2em;

  line-height: 24px;
  overflow: hidden;
  position: relative;

  background: ${props => props.theme.palette.background.secondary};

  border-radius: 12px;
  box-shadow: 0 0 20px ${props => props.theme.palette.shadow.main}, 0 0px 40px ${props => props.theme.palette.shadow.secondary};
`;

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  padding: 2em 0;

  .budget_edit__btn {
    height: 1em;
    margin-top: 1em;
  }
`;
