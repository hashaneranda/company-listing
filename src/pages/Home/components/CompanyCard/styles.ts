import styled from 'styled-components';
import tinycolor from 'tinycolor2';

export const Card = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 40em;
  padding: 1em 2em;

  line-height: 24px;
  overflow: hidden;
  position: relative;

  border-radius: 12px;
  box-shadow: 0 0 20px ${props => props.theme.palette.shadow.main}, 0 0px 40px ${props => props.theme.palette.shadow.secondary};

  ${props => props.theme.title === 'dark' && `background: ${props.theme.palette.background.secondary}`};

  &:hover {
    cursor: pointer;
    background: ${props => props.theme.palette.background.secondary};

    ${props =>
      props.theme.title === 'dark' && `background: ${tinycolor(props.theme.palette.background.secondary).lighten(5).toHexString()}`};
  }
`;

export const CardHeader = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  p {
    color: ${props => props.theme.palette.typography.secondary};
  }

  .date_container {
    display: flex;
    flex-direction: column;
    text-align: right;
  }
`;

export const CardBody = styled.div`
  display: grid;
  grid-template-rows: auto 1fr auto;
  text-align: center;
`;
