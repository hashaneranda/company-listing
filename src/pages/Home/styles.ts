import styled from 'styled-components';

export const Container = styled.div`
  padding: 1em 2em;
  display: grid;
  place-items: center;
  min-height: 100vh;
`;

export const MainWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
`;

export const CompanyWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 2em;

  margin-top: 3em;
`;
