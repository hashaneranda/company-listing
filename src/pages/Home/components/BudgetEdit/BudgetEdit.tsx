import React from 'react';
import { useFormik } from 'formik';
import * as Yup from 'yup';

// assets
import { budgetIcon } from 'config/images';

// components
import { TextInput } from 'common/components/FormHelper/FormHelper';
import { PrimaryButton } from 'common/components/Button/Button';
import { successNoty } from 'common/components/Notification/Notification';

import { Container, Form } from './styles';

interface CompanyProps {
  data: any;
  handleSubmit: (e: any) => void;
}

interface InitialData {
  [index: string]: number | string;
}

export const initialData: InitialData = {
  budget: '',
};

const BudgetEdit: React.FC<CompanyProps> = ({ data, handleSubmit }) => {
  const validationSchema = Yup.object().shape({
    budget: Yup.number().required('Budget is Required').min(data.budget_spent),
  });

  const formik = useFormik({
    initialValues: initialData,
    validationSchema: validationSchema,
    onSubmit: (v: InitialData) => {
      const payload = {
        ...data,
        budget: v.budget,
      };

      handleSubmit(payload);
      successNoty({ msg: `${data?.name} budget updated successfully` });
    },
  });

  return (
    <Container>
      <h3>{data?.name}</h3>
      <Form onSubmit={formik.handleSubmit}>
        <TextInput
          startAdorment={<img src={budgetIcon.default} alt='budget' />}
          label='Budget'
          id='budget'
          name='budget'
          type='text'
          onChange={formik.handleChange}
          value={formik.values['budget']}
          onBlur={formik.handleBlur}
          inputError={formik.errors['budget']}
        />

        <PrimaryButton className='budget_edit__btn' type='submit' data-testid='submit_btn'>
          Update
        </PrimaryButton>
      </Form>
    </Container>
  );
};

export default BudgetEdit;
