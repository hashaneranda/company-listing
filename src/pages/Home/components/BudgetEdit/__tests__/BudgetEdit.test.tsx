import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react';

import BudgetEdit from '../BudgetEdit';

import { StylesProvider } from '@material-ui/core/styles';
import Theme from 'theme/theme';
import ThemeSwitcher from 'theme/themeSwitcher';

const renderWithStyles = (component: React.ReactNode) => {
  return render(
    <StylesProvider injectFirst>
      <ThemeSwitcher>
        <Theme>{component}</Theme>
      </ThemeSwitcher>
    </StylesProvider>,
  );
};

const sampleData = {
  id: 1,
  name: 'Martian Firma',
  budget: 10000.0,
  budget_spent: 4500.0,
  date_of_first_purchase: '2119-07-07',
};

describe('BudgetEdit Componenet', () => {
  it('should take a snapshot', () => {
    const handleSubmit = jest.fn();

    const { asFragment } = renderWithStyles(<BudgetEdit data={sampleData} handleSubmit={handleSubmit} />);

    expect(asFragment(<BudgetEdit data={sampleData} handleSubmit={handleSubmit} />)).toMatchSnapshot();
  });

  it('should submit form with corrent values', async () => {
    const handleSubmit = jest.fn();
    const { getByTestId, container } = renderWithStyles(<BudgetEdit data={sampleData} handleSubmit={handleSubmit} />);

    const budgetInput = container.querySelector('input[name="budget"]');
    fireEvent.change(budgetInput, { target: { value: 50000 } });

    fireEvent.click(getByTestId('submit_btn'));

    await waitFor(() =>
      expect(handleSubmit).toHaveBeenCalledWith({
        ...sampleData,
        budget: '50000',
      }),
    );
  });

  it('should show validation on blur', async () => {
    const handleSubmit = jest.fn();
    const { container } = renderWithStyles(<BudgetEdit data={sampleData} handleSubmit={handleSubmit} />);

    const budgetInput = container.querySelector('input[name="budget"]');
    fireEvent.blur(budgetInput);

    await waitFor(() => {
      expect(container.querySelector('.MuiFormHelperText-root')).not.toBe(null);
      expect(container.querySelector('.MuiFormHelperText-root')).toHaveTextContent('Budget is Required');
    });
  });
});
