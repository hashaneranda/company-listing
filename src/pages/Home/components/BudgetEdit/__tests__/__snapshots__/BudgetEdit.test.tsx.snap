// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetEdit Componenet should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-jRQAMF VWGDG"
  >
    <h3>
      Martian Firma
    </h3>
    <form
      class="sc-gKckTs jrnvto"
    >
      <div
        class="MuiFormControl-root sc-bdvvaa iYNPUu"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
            data-shrink="true"
            for="budget"
            id="budget-label"
          >
            Budget
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <img
                alt="budget"
                src="money-bag.svg"
              />
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
              id="budget"
              name="budget"
              type="text"
              value=""
            />
          </div>
        </div>
        <p
          class="MuiFormHelperText-root defaultHellperTxt"
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text sc-eCImvq ueBvO budget_edit__btn"
        data-testid="submit_btn"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Update
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
